### ISSUE TITLE
Idea/wish: "ROM directory"

### ISSUE THREAD

**Issue by lgblgblgb** [commanderx16/x16-rom#13]
Just an idea ... As X16 seems to have more ROM pages in a 8K window, it can be expected that there can be custom ROM contents as well in the future, and also more and more stuffs in the "official ROM pack" as well.

Some 8 bit machines (I can mention the Z80 based Enterprise-128, but I guess Acorn systems as well - I am not sure about the second, but for example to stay with Commodore, the very rare Commodore LCD too) there is way to list ROMs, also to use them easily, if there is an entry point for that.

My idea is to put some identifier string into ROMs which marks some kind of "ROM directory" with one or more entry points listed (a ROM can have multiple entries as well), with some short name to easy to refer them. So then, there can be even a BASIC token allocated to use a search+list routine to list those for the user without an argument, or with an argument, to give the control to that functionality.

In this way, some can have eg a menu based program loader, which can be quickly call that, without the need to first load it from disk. And maybe other usages as well, it would be nice.

The alternative of course, to know your system and use various SYS commands with the right parameter :) [I'm not sure if there is SYS btw, for paged ROM to also select which ROM "page" should be used to utilize the jump there].

**Comment by mist64** [comment]
Yes, every ROM page should have a common structure. I'm thinking a few bytes at the very end of the page.

**Comment by lgblgblgb** [comment]
Sounds cool, and btw, there can be (optional) init _and_ startup points for each items, so KERNAL can even initialize each (by calling the startup init func), if ROMs provides entry point for "startup init" not only the "user call". I got this idea from my disassembling works on the Commodore LCD's kernal btw, and I like the approach since then :)

**Comment by BruceMcF** [comment]
After reviewing the PRG, it would appear that anywhere in $FFxx that is not used for interrupt handling by the processor is fair game for the Kernel jump table. So the simplest place to put a ROM signature is the top eight bytes of $FExx, $FFE8 to $FFEF. To allow using all eight for name, put a nul in trailing bytes of the name that are not used.
